fabric-fluid-v1/src/main/java/net/fabricmc/fabric/api/fluid/v1/FabricFlowableFluid.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.fluid.v1;
+
+import java.util.Optional;
+
+import net.minecraft.client.render.BackgroundRenderer;
+import net.minecraft.client.render.FogShape;
+import net.minecraft.client.world.ClientWorld;
+import net.minecraft.entity.Entity;
+import net.minecraft.fluid.FlowableFluid;
+import net.minecraft.sound.SoundEvent;
+
+/**
+ * Extends the {@link FlowableFluid} class by adding more features.
+ */
+public abstract class FabricFlowableFluid extends FlowableFluid {
+	/**
+	 * Gets the color of the fluid fog.
+	 *
+	 * @param player    Player submerged by the fluid.
+	 * @param tickDelta Time passed from the last tick.
+	 * @param world     The current client-side world.
+	 * @return <p>An int value indicating the color of the fluid fog.</p>
+	 * <p>If the color is -1 no fog will be rendered.</p>
+	 * <p>(It could be a hexadecimal value like 0xFFFFFF).</p>
+	 */
+	public abstract int getFabricFogColor(Entity player, float tickDelta, ClientWorld world);
```

Comment:
Just like the previous approach, this will crash on dedicated servers because it references client-only code in common code.

Since the fog stuff is related to rendering, I'd add them as new methods in `FluidRenderHandler` which seems to be the simplest option. Doesn't require any new classes either.